### PR TITLE
Add unit test for StoreToJobLister

### PR DESCRIPTION
**Pull request by mqliang**
This PR add unit test for StoreToJobLister

### CODE REVIEW

**Comment by ashcrow** [comment]
@mqliang can you squash the two commits into one since the second is fixing gofmt found items?

**Comment by mqliang** [comment]
`Shippable` result is very confusing. Who can help to point out why it's not successful.

**Comment by ashcrow** [comment]
@mqliang I believe it is a flake. The file in question is actually present in your tree. Try closing and opening the PR to get shippable to try again.

**Comment by ashcrow** [comment]
Very odd. Same error happened yet you didn't modify the file and it is in your tree. I'm going to see if this is happening with any other PRs as well.

**Comment by mqliang** [comment]
@ashcrow Thanks

**Comment by ashcrow** [comment]
@mqliang no problem!!

File: pkg/client/cache/listers_test.go [hunk 1]
```diff
@@ -278,6 +278,128 @@ func TestStoreToDaemonSetLister(t *testing.T) {
...
+			store.Add(r)
+		}
+
+		daemonSets, err := c.list()
+		if err != nil && c.expectErr {
+			continue
+		} else if c.expectErr {
+			t.Fatalf("Expected error, got none")
+		} else if err != nil {
+			t.Fatalf("Unexpected error %#v", err)
```
Comment (krousey) [inline pkg/client/cache/listers_test.go]: These fatals will halt the test function entirely, and subsequent test cases won't run. Could you use `Errorf` and `continue` instead?

File: pkg/client/cache/listers_test.go [hunk 2]
```diff
@@ -278,6 +278,128 @@ func TestStoreToDaemonSetLister(t *testing.T) {
...
+		}
+		daemonSetNames := make([]string, len(daemonSets))
+		for ix := range daemonSets {
+			daemonSetNames[ix] = daemonSets[ix].Name
+		}
+		if !c.outJobNames.HasAll(daemonSetNames...) || len(daemonSetNames) != len(c.outJobNames) {
+			t.Errorf("Unexpected got controllers %+v expected %+v", daemonSetNames, c.outJobNames)
```
Comment (krousey) [inline pkg/client/cache/listers_test.go]: There seems to be no easy way to determine which test case failed from this error message, apart from hoping that c.outJobNames uniquely identifies a test case.

File: pkg/client/cache/listers_test.go [hunk 3]
```diff
@@ -278,6 +278,128 @@ func TestStoreToDaemonSetLister(t *testing.T) {
...
+	for _, c := range testCases {
+		for _, r := range c.inJobs {
+			store.Add(r)
+		}
+
+		daemonSets, err := c.list()
+		if err != nil && c.expectErr {
+			continue
+		} else if c.expectErr {
+			t.Fatalf("Expected error, got none")
```
Comment (krousey) [inline pkg/client/cache/listers_test.go]: Expected an error on which test case? How would I track this down if I saw it in the log?

Comment (krousey) [inline pkg/client/cache/listers_test.go]: Also, there's no need to use `Fatalf` if there's no formatting.

File: pkg/client/cache/listers_test.go (hunk 1 above)
Comment (mqliang) [inline pkg/client/cache/listers_test.go]: @krousey Thanks

**Comment by mqliang** [comment]
I closed this PR and then force-pushed, so I can not reopen it. I recommit the modification in another PR: https://github.com/kubernetes/kubernetes/pull/15753

I am sorry the inconvenience bring to you. BTW, thanks a lot for all of your help and review.

**Pull request closed without merging**